hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestApacheClientConnectionPool.java

```diff
@@ -60,4 +74,36 @@ public void testKacIsClosed() throws Throwable {
       verifyCause(ClosedIOException.class, ex);
     }
   }
+
+  @Test
+  public void testConnectionLogging() throws Exception {
```

Comment:
I'd like this split into two test cases, for better isolation. You could factor out the common stuff with the connection creation being the only difference
* httpRouteConnection
* validation of the connection

Reply:
Taken. Have broken into two tests:
* testNonConnectedConnectionLogging
* testConnectedConnectionLogging